test/correctness/float16_t.cpp

```diff
@@ -290,43 +307,71 @@ int main(int argc, char **argv) {
             float pos_inf = std::numeric_limits<float>::infinity();
             float16_t fp16_pos_inf(pos_inf);
             if (fp16_pos_inf != float16_t::make_infinity()) {
-                printf("Conversion of 32-bit positive infinity to 16-bit float is %x, not positive infinity.\n", fp16_pos_inf.to_bits());
+                fprintf(stderr, "Conversion of 32-bit positive infinity to 16-bit float is %x, not positive infinity.\n", fp16_pos_inf.to_bits());
                 return -1;
             }
 
             float neg_inf = -std::numeric_limits<float>::infinity();
             float16_t fp16_neg_inf(neg_inf);
             if (fp16_neg_inf != float16_t::make_negative_infinity()) {
-                printf("Conversion of 32-bit negative infinity to 16-bit float is %x, not negative infinity.\n", fp16_neg_inf.to_bits());
+                fprintf(stderr, "Conversion of 32-bit negative infinity to 16-bit float is %x, not negative infinity.\n", fp16_neg_inf.to_bits());
                 return -1;
             }
 
             Param<float> f_in("f_in");
             f_in.set(pos_inf);
             c = evaluate<float16_t>(cast(Float(16), f_in));
             if (c != float16_t::make_infinity()) {
-                printf("Halide conversion of 32-bit positive infinity to 16-bit float is %x, not positive infinity.\n", c.to_bits());
+                fprintf(stderr, "Halide conversion of 32-bit positive infinity to 16-bit float is %x, not positive infinity.\n", c.to_bits());
                 return -1;
             }
 
             f_in.set(neg_inf);
             c = evaluate<float16_t>(cast(Float(16), f_in));
             if (c != float16_t::make_negative_infinity()) {
-                printf("Halide conversion of 32-bit negative infinity to 16-bit float is %x, not negative infinity.\n", c.to_bits());
+                fprintf(stderr, "Halide conversion of 32-bit negative infinity to 16-bit float is %x, not negative infinity.\n", c.to_bits());
                 return -1;
             }
         }
     }
 
+    return 0;
+}
+
+}  // namespace
+
+int main(int argc, char **argv) {
+    MyCustomErrorReporter reporter;
+    set_custom_compile_time_error_reporter(&reporter);
+
+    printf("Testing float16_t...\n");
+    if (run_test<float16_t>() != 0) {
+        fprintf(stderr, "float16_t test failed!\n");
+        return -1;
+    }
+
+    printf("Testing _Float16...\n");
 #ifdef HALIDE_CPP_COMPILER_HAS_FLOAT16
+    if (run_test<_Float16>() != 0) {
+        fprintf(stderr, "_Float16 test failed!\n");
+        return -1;
+    }
+
+#ifdef __clang__
     {
         float16_t f(1.0f16);
         _Float16 f2 = (_Float16)f;
         if (f2 != 1.0f16) {
-            printf("Roundtrip of 16-bit float via _Float16 failed.\n");
+            fprintf(stderr, "Roundtrip of 16-bit float via _Float16 failed.\n");
             return -1;
         }
     }
+#else
+    printf("Only clang supports _Float16 literals, skipping roundtrip test\n");
```

Comment:
I can't get this. In GCC, we can't use `_Float16` even if `HALIDE_CPP_COMPILER_HAS_FLOAT16` is set?

Reply:
As far as I can tell, even the versions of gcc that support `_Float16` don't support 16-bit-floating-point-constants via the `f16` suffix. (Maybe I don't have it configured correctly, but I also can't find any gcc docs that suggest that it does.)

Reply:
Got it. Thanks for the explanation.